addons/sale/controllers/catalog.py

```diff
@@ -0,0 +1,112 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route, Controller
+from odoo.tools import groupby
+
+
+class CatalogController(Controller):
+
+    @route('/sales/catalog/sale_order_lines_info', auth='user', type='json')
+    def sale_product_catalog_get_sale_order_lines_info(self, order_id, product_ids):
+        """ Returns products information to be shown in the catalog.
+
+        :param int order_id: The sale order, as a `sale.order` id.
+        :param list product_ids: The products currently displayed in the product catalog, as a list
+                                 of `product.product` ids.
+        :rtype: dict
+        :return: A dict with the following structure:
+            {
+                'productId': int
+                'quantity': float (optional)
+                'price': float
+                'readOnly': bool (optional)
+            }
+        """
+        order = request.env['sale.order'].browse(order_id)
+        sale_order_line_info = []
+        for product, lines in groupby(
+            order.order_line.filtered(lambda line: not line.display_type),
+            lambda line: line.product_id
+        ):
+            if (len(lines) > 1):
+                price_unit = lines[0].order_id.pricelist_id._get_product_price(
+                    product=lines[0].product_id.with_context(
+                        **lines[0]._get_product_price_context()
+                    ),
+                    quantity=1.0,
+                    currency=lines[0].order_id.currency_id,
+                    date=lines[0].order_id.date_order,
+                )
```

Comment:
At this point, I need it because the uom can be different for each product, and also because I don't show the quantity so I assumed the price for 1 should be showed.